### ISSUE TITLE
Async refinements

### ISSUE THREAD

**Issue by BlessedRaccoon** [RobinTail/express-zod-api#269]
Hello,

First of all, thank you very much for your package!

I'm starting using it and I'm running into an issue. I have a piece of code that looks like this:
```ts
input: z.object({
        username: z.string().nonempty().max(100).refine(async username => {
            return await someAsyncFunction(username} == 0
       })
}),
```
I'm doing an async refinement, which should be allowed according to Zod docs. But I am getting this error:
`Async refinement encountered during synchronous parse operation. Use .parseAsync instead.`.

It looks like my `username` is not being parsed with parseAsync() but with parse(). Is the fact that we can't use async refinements on purpose?

**Comment by RobinTail** [comment]
Thank you for your feedback @BlessedRaccoon .
You made a very good finding. I think this error is related to the usage of `.parse` within `Endpoint`.
I'll try to replace it with `.parseAsync()`.
Thank you.

**Comment by RobinTail** [comment]
@BlessedRaccoon ,
the issue has been fixed in the PR #270 and the fix has been released in version 5.3.1 🚀 
Please upgrade.
In case it persists, please create another issue.
Thank you for your contribution!
